### PR TITLE
Add tutorial to use debugging in vscode for unit tests

### PR DESCRIPTION
**Pull request by takameyer**
## What, How & Why?
I was able to get LLDB working in vscode with help from @fronck and @cesarvr.  I decided it would be good to document this process and add it to our contribution docs.

## ☑️ ToDos

* [ ] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 🚦 Tests
* [ ] 📝 Public documentation PR created or is not necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [ ] typescript definitions file is updated
* [ ] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

**Review by kneth — APPROVED** [review]

**Review by cesarvr — APPROVED** [review]
LGTM

File: .vscode/launch.json [hunk 1]
```diff
@@ -75,7 +75,20 @@
...
+        {
+            "type": "lldb",
+            "request": "launch",
+            "name": "LLDB Launch Unit Tests",
+            "program": "node",
+            "args": [
+                "--expose_gc",
+                "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",
+                "spec/unit_tests.js",
+                "--filter=testListPush"
```
Comment (fronck) [inline .vscode/launch.json:88]:
```suggestion
                "--filter=."
```

File: contrib/vscode-debugging.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+Then you will need the following plugins for VSCode:
+* [C/C++ by Microsoft](https://github.com/Microsoft/vscode-cpptools)
+* [CodeLLDB](https://github.com/vadimcn/vscode-lldb)
+
+## Building
+
+In order to debug C++, you must first build the project in debug mode.  This can be done with the command
+
+```bash
+$ npm run rebuild --debug
```
Comment (fronck) [inline contrib/vscode-debugging.md:16]: This won't work on v11+ (`develop`), will it?  We should state that here or provide alternative instructions.

File: contrib/vscode-debugging.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        "--expose_gc",
+        "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",
+        "spec/unit_tests.js",
+        "--filter=."
+    ],
+    "cwd": "${workspaceFolder}/tests"
+}
+```
+
+A quick read through this code shows that the launch type is `lldb` and it is using the `node` command to invoke the `jasmine` test framework with our `spec/unit_tests.js`.  The filter option is currently running all tests.  To make things easier, lets modify that param to take a single test (`testListPush` from `tests/list-tests.js`).
```
Comment (fronck) [inline contrib/vscode-debugging.md:41]:
```suggestion
A quick read through this code shows that the launch type is `lldb` provided by the CodeLLDB extension, and it is using the `node` command to invoke the `jasmine` test framework with our `spec/unit_tests.js`.  The filter option is currently running all tests.  To make things easier, let's modify that parameter to take a single test (`testListPush` from `tests/list-tests.js`).
```

File: contrib/vscode-debugging.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+```json
+{
+    "type": "lldb",
+    "request": "launch",
+    "name": "LLDB Launch Unit Tests",
+    "program": "node",
+    "args": [
+        "--expose_gc",
+        "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",
```
Comment (fronck) [inline contrib/vscode-debugging.md:51]:
```suggestion
        "${workspaceFolder}/tests/node_modules/jasmine/bin/jasmine.js",
```

File: contrib/vscode-debugging.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
```
Comment (fronck) [inline contrib/vscode-debugging.md:33]:
```suggestion
        "${workspaceFolder}/tests/node_modules/jasmine/bin/jasmine.js",
```

File: .vscode/launch.json [hunk 6]
```diff
@@ -75,7 +75,20 @@
...
-        }
+        },
+        {
+            "type": "lldb",
+            "request": "launch",
+            "name": "LLDB Launch Unit Tests",
+            "program": "node",
+            "args": [
+                "--expose_gc",
+                "${workspaceRoot}/tests/node_modules/jasmine/bin/jasmine.js",
```
Comment (fronck) [inline .vscode/launch.json:86]:
```suggestion
                "${workspaceFolder}/tests/node_modules/jasmine/bin/jasmine.js",
```

**Review by fronck — CHANGES_REQUESTED** [review]
`$workspaceRoot` in `launch.json` is deprecated in favor `$workspaceFolder`, so I think we should change that.

Other than that, just a few suggestions :)

File: contrib/vscode-debugging.md (hunk 2 above)
Comment (takameyer) [reply contrib/vscode-debugging.md:16]: I did test this in develop actually.

**Review by takameyer — COMMENTED** [review]
LGTM

**Review by fronck — APPROVED** [review]
:+1:

**Pull request merged**